menu.cpp

```diff
@@ -64,7 +88,25 @@ void test_menu()
   // could I use try and catch framework for exceptions in this case?
   {
     menu v;
-    assert(v.get_button(0).get_name() == "action");
+    assert(v.get_button(0).get_label() == "action");
+  }
+  {
+    // (482) Menu button can be called from its label
+    menu m;

```

Comment:
Ideally, it should be `const menu v`, as you only read from the menu. I see why: `get_button` is non-const, which is reasonable enough. There is a solution, however, which is *adding* a const version:

```c++
class menu_view
{
  // ...
  button& get_button(const int index); // Currently there, signature and name probably incorrect

  const button& get_button(const int index) const; // To be added
}
```

Because I think you'll enjoy that new piece of knowledge, I will not approve this Pull Request, so you can improve even more on your const-correctness :-)

Reply:
On it!
But why does one need to add `const` both before and after the declaration?

Reply:
Ah, helaas pindakaas I can't get the function to return a `const` output. I think one would need to make `m_v_buttons` a const variable as well, but that's not what a member variable should be.
I'm very confused about how the compiler deals with const, I'll just ignore this and move on for now.